### DISCUSSION TITLE
Gunicorn vs uWSGI

### DISCUSSION THREAD

**Discussion by rajan-code** [miguelgrinberg/Flask-SocketIO#2008] [General]
If I understand the [documentation](https://flask-socketio.readthedocs.io/en/latest/deployment.html) correctly, if nginx is configured correctly and a message queue is used, both gunicorn and uwsgi support multiple workers right?  
Let's say I have no more than 100 websocket concurrent connections. Does uwsgi have any advantages that would make it worth using?

Also, the docs say `The workaround to use multiple worker processes with gunicorn is to launch several single-worker instances`
does this mean simply set number of workers to a num greater than 1 e.g. `gunicorn -w 2 --threads 100 module:app` ?

**Comment by miguelgrinberg** [comment]
@rajan-code first of all, for 100 connections a single worker is very likely more than enough. I personally choose gunicorn over uwsgi.

Launching several single worker processes means exactly that. You run `gunicorn -w 1 -b : 5000 ...` and then `gunicorn -w 1 -b :5001 ...` and so on, as many times as you need.
